lightningd/gossip_control.c

```diff
@@ -357,9 +357,10 @@ static void json_listchannels_reply(struct subd *gossip UNUSED, const u8 *reply,
 					       &entries[i].short_channel_id));
 		json_add_bool(response, "public", entries[i].public);
 		json_add_u64(response, "satoshis", entries[i].satoshis);
-		json_add_num(response, "flags", entries[i].flags);
+		json_add_num(response, "message_flags", entries[i].message_flags);
+		json_add_num(response, "channel_flags", entries[i].channel_flags);
```

Comment:
This is a JSON API change, so you need to do something like:

```
/* Prior to spec v0891374d47ddffa64c5a2e6ad151247e3d6b7a59, these two were a single u16 field */
if (deprecated_apis)
        json_add_num(response, "flags", ((u16)entries[i].message_flags << 8) | entries[i].channel_flags);
```

Reply:
Also implies we need CHANGELOG.md update, but that means we need a separate commit which adds a new section to CHANGELOG.md.  I'll push that now...

Reply:
I'm assuming it's ok to add fields to the JSON api, it's just removing them that's a no-go?

Reply:
I also updated the CHANGELOG, lmk if you had something else in mind!

Reply:
Perfect!!